share/build/buildlib.spio

```diff
@@ -86,80 +92,75 @@ def buildlib(bldroot, installpath, case):
+    if not os.path.isdir(pio_bld_dir):
+        os.makedirs(pio_bld_dir)
+
+    # Compute cmake args
+    # Use old genf90 until "short" type is supported
+    cmake_opts = "-Wno-dev "
+    cmake_opts = f"-DGENF90_PATH={scorpio_src_dir}/src/genf90 "
+    if "ADIOS2_ROOT" in os.environ:
+        cmake_opts += "-DWITH_ADIOS2:BOOL=ON "
+    if debug:
+        cmake_opts += "-DPIO_ENABLE_LOGGING=ON "
+    # Case changes for NetCDF/NETCDF forces us to do this. For other packages
+    # what's already in the env should work
+    if "NETCDF_PATH" in os.environ:
+        cmake_opts += f"-DNetCDF_PATH:PATH={os.environ['NETCDF_PATH']} "
+    if "NETCDF_C_PATH" in os.environ:
+        cmake_opts += f"-DNetCDF_C_PATH:PATH={os.environ['NETCDF_C_PATH']} "
+    if "NETCDF_FORTRAN_PATH" in os.environ:
+        cmake_opts += f"-DNetCDF_Fortran_PATH:PATH={os.environ['NETCDF_FORTRAN_PATH']} "
+    if "PNETCDF_PATH" in os.environ:
+        cmake_opts += f"-DPnetCDF_PATH:PATH={os.environ['PNETCDF_PATH']} "
     else:
-        cmake_opts = '"-D GENF90_PATH=' + srcroot + '/libraries/parallelio/scripts/ "'
-
-    stdargs = get_standard_makefile_args(case, shared_lib=True)
-
-    gmake_vars = (
-        "CASEROOT={caseroot} COMP_NAME={pio_model} "
-        "USER_CMAKE_OPTS={cmake_opts} "
-        "PIO_LIBDIR={pio_dir} CASETOOLS={casetools} "
-        "USER_CPPDEFS=-DTIMING".format(
-            caseroot=caseroot,
-            pio_model=pio_model,
-            cmake_opts=cmake_opts,
-            pio_dir=pio_dir,
-            casetools=casetools,
-        )
-    )
-
-    if scorpio_src_root_dir is not None:
-        gmake_vars += (
-            " IO_LIB_SRCROOT={scorpio_src_root_dir} "
-            " IO_LIB_v1_SRCDIR={scorpio_classic_dir} "
-            " IO_LIB_v2_SRCDIR={scorpio_dir} ".format(
-                scorpio_src_root_dir=scorpio_src_root_dir,
-                scorpio_classic_dir=scorpio_classic_dir,
-                scorpio_dir=scorpio_dir,
-            )
-        )
-
-    gmake_opts = (
-        "{pio_dir}/Makefile -C {pio_dir} "
-        " {gmake_vars} {stdargs} -f {casetools}/Makefile".format(
-            pio_dir=pio_dir, gmake_vars=gmake_vars, casetools=casetools, stdargs=stdargs
-        )
-    )
+        cmake_opts += "-DWITH_PNETCDF:LOGICAL=FALSE -DPIO_USE_MPIIO:LOGICAL=FALSE "
+    if mpilib == "mpi-serial":
+        cmake_opts += f"-DPIO_USE_MPISERIAL=TRUE -DMPISERIAL_PATH={installpath} "
+    cmake_opts += "-DCMAKE_VERBOSE_MAKEFILE:BOOL=ON "
+    cmake_opts += f"-DGPTL_PATH:STRING={installpath} "
+    cmake_opts += "-DPIO_ENABLE_TESTS:BOOL=OFF -DPIO_USE_MALLOC:BOOL=ON "
+    cmake_opts += f"-DUSER_CMAKE_MODULE_PATH:LIST={cimeroot}/CIME/non_py/src/CMake "
+
+    # Hacky way of getting stuff from Macros. Should scorpio just load the E3SM
+    # macros once they are using cmake names for things?
+    make_args = get_standard_makefile_args(case, shared_lib=True)
+    macro_dump = run_cmd(f"make -f Macros.make COMP_NAME=spio {make_args} -p")[1]
```

Comment:
Why can't we just use `Macros.cmake` directly?

Reply:
The macros don't use Cmake names for things, so would have no effect on the spio build.

Reply:
But aren't we going to "abandon" support for `Macros.make` at some point? In that sense, wouldn't it be more long-term planning to parse `Macros.cmake`? I know var names are not the same, I wasn't suggesting to feed it straight to scorpio, but just to parse that instead of the makefile one.